C/services/south/south.cpp

```diff
@@ -940,6 +959,86 @@ void SouthService::configChange(const string& categoryName, const string& catego
 	}
 }
 
+/**
+ * Separate thread to run plugin_reconf, to avoid blocking 
+ * service's management interface due to long plugin_poll calls
+ */
+static void reconfThreadMain(void *arg)
+{
+	SouthService *ss = (SouthService *)arg;
+	Logger::getLogger()->info("reconfThreadMain(): Spawned new thread for plugin reconf");
+	while(1)
+	{
+		ss->handlePendingReconf();
+	}
+	Logger::getLogger()->info("reconfThreadMain(): plugin reconf thread exiting");
+}
+
+/**
+ * Handle configuration change notification
+ */
+void SouthService::handlePendingReconf()
+{
+	Logger::getLogger()->debug("SouthService::handlePendingReconf: Going into cv wait");
+	mutex mtx;
+	unique_lock<mutex> lck(mtx);
+	m_cvNewReconf.wait(lck);
+	Logger::getLogger()->debug("SouthService::handlePendingReconf: cv wait has completed");
+
+	while(1)
+	{
+		unsigned int numPendingReconfs = 0;
+		{
+			lock_guard<mutex> guard(m_pendingNewConfigMutex);
+			numPendingReconfs = m_pendingNewConfig.size();
+			if (numPendingReconfs)
+				Logger::getLogger()->debug("SouthService::handlePendingReconf(): will process %d entries in m_pendingNewConfig", numPendingReconfs);
+			else
+			{
+				Logger::getLogger()->debug("SouthService::handlePendingReconf DONE");
+				break;
+			}
+		}
+
+		for (unsigned int i=0; i<numPendingReconfs; i++)
```

Comment:
If you have more than one pending reconfiguration you don't need to worry about it, only the last one. Each reconfiguration request comes with a complete configuration category.

Reply:
I am thinking that sometimes reconf could have side effects and hence it is probably better to process all reconf requests in order. e.g. a device register value set from 1->2->3 might indicate something special, etc.